common/rfb/CMsgWriter.cxx

```diff
@@ -182,13 +182,28 @@ void CMsgWriter::writePointerEvent(const Point& pos, int buttonMask)
   if (p.x >= server->width()) p.x = server->width() - 1;
   if (p.y >= server->height()) p.y = server->height() - 1;
 
+  if (server->supportsExtendedMouseButtons)
+  {
+    writePointerEventExt(pos,buttonMask);
+    return;
+  }
+
   startMsg(msgTypePointerEvent);
   os->writeU8(buttonMask);
   os->writeU16(p.x);
   os->writeU16(p.y);
   endMsg();
 }
 
+void CMsgWriter::writePointerEventExt(const Point& p, int buttonMask)
+{
+  startMsg(msgTypePointerEventExt);
+  os->writeU16(buttonMask);
+  os->writeU16(p.x);
+  os->writeU16(p.y);
+  endMsg();
+}
```

Comment:
for sure. I had it in different methods to lessen merge conflicts since i was just running this locally. it makes sense to make the main method handle both cases if it's going to be in master.